src/components/Modal/Modal.tsx

```diff
@@ -1,161 +1,42 @@
-import * as React from 'react';
+import React from 'react';
 import classnames from 'classnames';
 import styles from './styles.module.scss';
-import {SquareModal, StripeModal} from '../ModalPayment';
-
-import { Elements } from '@stripe/react-stripe-js';
-import { loadStripe } from '@stripe/stripe-js';
+import ModalAmount from '../ModalAmount';
+import {SquareModal} from '../ModalPayment';
+import ModalConfirmation from '../ModalConfirmation';
+import { useModalPaymentState, useModalPaymentDispatch } from "../../utilities/hooks/ModalPaymentContext/context"
+import { CLOSE_MODAL } from "../../utilities/hooks/ModalPaymentContext/constants"
 
 interface Props {
   purchaseType: string;
-  handleClose: (event: React.MouseEvent<HTMLButtonElement>) => void;
-  show: boolean;
   sellerId: string;
   sellerName: string;
 }
 
-interface State {
-  amount: number;
-  customInput: boolean;
-  close: boolean;
-  next: boolean;
-  showBillModal: boolean;
-}
-
-const ModalPaymentBox: any = process.env.REACT_APP_USE_STRIPE === "TRUE" ? StripeModal : SquareModal ;
-
-const stripePK = process.env.REACT_APP_STRIPE_PUBLISHABLE_KEY!;
```

Comment:
This null assertion at compile time here is a little problematic since it can lead to run time errors if it fails to get the key for some reason. We should probably add a to do or something to come back and make this more fail safe.